### PR TITLE
Update to Keys and Certificate Types Section

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-ietf-teep-architecture.md [hunk 1]
```diff
@@ -735,11 +735,12 @@ Note that personalization data is not included in the table above.
...
-TEEP requests from a TAM to a TEEP Agent can be encrypted with the
-TEE public key (to provide confidentiality), and are then signed with the TAM
+TEEP requests from a TAM to a TEEP Agent can be signed with the TAM
 private key (for authentication and integrity protection).
-Conversely, TEEP responses from a TEEP Agent to a TAM can be encrypted
-with the TAM public key, and are then signed with the TEE private key.  
+Conversely, TEEP responses from a TEEP Agent to a TAM can be 
+signed with the TEE private key. For encryption of the personalization
+data and the TA binary, the TA developer has to use public keys 
+unique to the TEE.
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: "use" how?  This is confusing.  I prefer the old text, and line 745-746 only make sense with the old text, not this proposed change.

File: draft-ietf-teep-architecture.md [hunk 2]
```diff
@@ -798,12 +800,12 @@ for any other TA, as discussed in {{trust-anchors-in-teep-agent}}.
 The Trust Anchor Store in a TAM consists of a list of Trust Anchors, which
 are certificates that sign various device TEE certificates.  A TAM will accept a
 device for TA management if the TEE in the device uses a TEE certificate
-that is chained to a certificate that the TAM trusts.
+that is chained to a certificate that the TAM trusts, is contained in a whitelist, 
+is not found on a blacklist and fulfills other policy criteria.
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]:
```suggestion
that is chained to a certificate or raw public key that the TAM trusts, is contained in an allow list, 
is not found on a block list, and/or fulfills any other policy criteria.
```

File: draft-ietf-teep-architecture.md [hunk 3]
```diff
@@ -798,12 +800,12 @@ for any other TA, as discussed in {{trust-anchors-in-teep-agent}}.
...
 
 ## Scalability
 
-This architecture uses a PKI, although self-signed certificates are
-also permitted.  Trust Anchors exist on the devices to
+This architecture uses a PKI. Trust Anchors exist on the devices to
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: Why is this removing self-signed certificates?
```suggestion
This architecture uses a PKI (including self-signed certificates). Trust Anchors exist on the devices to
```

File: draft-ietf-teep-architecture.md [hunk 4]
```diff
@@ -780,12 +781,13 @@ document.
...
 device with a particular TEE, it must obtain a TAM
-certificate from a CA that is listed in the Trust Anchor Store of the TEEP Agent.
+certificate from a CA or the raw public key of a TAM that is listed in 
+the Trust Anchor Store of the TEEP Agent.
 
 ## Trust Anchors in a TEE {#trust-anchors-in-tee}
 
 A TEE determines whether TA binaries are allowed to execute by 
-verifying whether the TA's signer chains up to a certificate
+verifying whether their signature can be verified by certificate(s)
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]:
```suggestion
verifying whether their signature can be verified using certificate(s) or raw public key(s)
```

File: draft-ietf-teep-architecture.md (hunk 1 above)
Comment (dthaler) [reply draft-ietf-teep-architecture.md]: Russ commented: 

> Section 5 says: "... encrypted with the TAM public key ...". This is not correct. The TAM public key is used to establish the key that is then used for encryption and decryption. Further the wording is aimed at RSA, which is a key transport algorithm. Please make the wording accommodate key transport and key agreement approaches.

However, I don't think this is addressed yet.

**Review by dthaler — CHANGES_REQUESTED** [review]

File: draft-ietf-teep-architecture.md (hunk 1 above)
Comment (dthaler) [reply draft-ietf-teep-architecture.md]: Ming, can you do a wording suggestion here?

File: draft-ietf-teep-architecture.md [hunk 5]
```diff
@@ -798,12 +800,12 @@ for any other TA, as discussed in {{trust-anchors-in-teep-agent}}.
 The Trust Anchor Store in a TAM consists of a list of Trust Anchors, which
 are certificates that sign various device TEE certificates.  A TAM will accept a
 device for TA management if the TEE in the device uses a TEE certificate
-that is chained to a certificate that the TAM trusts.
+that is chained to a certificate or raw public key that the TAM trusts, is contained in an allow list, 
+is not found on a block list, and/or fulfills any other policy criteria.
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:839]: Tiru asks: should we add a reference somewhere to an RFC that talks about the raw public key?

File: draft-ietf-teep-architecture.md (hunk 3 above)
Comment (dthaler) [reply draft-ietf-teep-architecture.md]: Also need to reword somehow to accommodate raw public key

File: draft-ietf-teep-architecture.md (hunk 5 above)
Comment (dthaler) [reply draft-ietf-teep-architecture.md:839]: Tiru asks: should (or does?) sec consid section talk about compromise of raw public key trust anchor?

File: draft-ietf-teep-architecture.md [hunk 6]
```diff
@@ -735,11 +735,12 @@ Note that personalization data is not included in the table above.
 The use of personalization data is dependent on how TAs are used 
 and what their security requirements are. 
 
-TEEP requests from a TAM to a TEEP Agent can be encrypted with the
-TEE public key (to provide confidentiality), and are then signed with the TAM
+TEEP requests from a TAM to a TEEP Agent can be signed with the TAM
```
Comment (mingpeiwk) [inline draft-ietf-teep-architecture.md]:
```suggestion
TEEP requests from a TAM to a TEEP Agent can be encrypted with a data key that is wrapped with
the TEE public key (to provide confidentiality), and are then signed with the TAM
```

File: draft-ietf-teep-architecture.md [hunk 7]
```diff
@@ -735,11 +735,12 @@ Note that personalization data is not included in the table above.
...
 private key (for authentication and integrity protection).
-Conversely, TEEP responses from a TEEP Agent to a TAM can be encrypted
-with the TAM public key, and are then signed with the TEE private key.  
+Conversely, TEEP responses from a TEEP Agent to a TAM can be
```
Comment (mingpeiwk) [inline draft-ietf-teep-architecture.md]:
```suggestion
Conversely, a TEEP response from a TEEP Agent to a TAM can be encrypted with a data key that is transported with the TAM public key, and is
```

File: draft-ietf-teep-architecture.md (hunk 1 above)
Comment (mingpeiwk) [reply draft-ietf-teep-architecture.md]: Revised with suggestions. I kept the encryption clause for its purpose to provide end-to-end security without relying on network security such as use of TLS to exchange the TEEP messages.

File: draft-ietf-teep-architecture.md (hunk 7 above)
Comment (hannestschofenig) [reply draft-ietf-teep-architecture.md]: We are not providing encryption capabilities for the communication from the TEEP Agent to the TAM.

File: draft-ietf-teep-architecture.md (hunk 5 above)
Comment (hannestschofenig) [reply draft-ietf-teep-architecture.md:839]: I added the term raw public key to the terminology section. The compromise of a raw public key trust anchor is not different from the compromise of a "certificate" trust anchor.

**Comment by dthaler** [comment]
FYI the "allow/block list" discussion in the industry made slashdot this week: https://developers.slashdot.org/story/20/06/14/1722223/github-android-python-go-more-software-adopts-race-neutral-terminology

**Review by dthaler — APPROVED** [review]

**Review by dthaler — APPROVED** [review]
Merging since Hannes (as author) and I approve the latest version.  I sent email 24 hours ago and heard no objections.  If further changes are needed, they can come in a subsequent PR.

File: draft-ietf-teep-architecture.md (hunk 7 above)
Comment (mingpeiwk) [reply draft-ietf-teep-architecture.md]: It was the original design that TEEP Agent and TAM can provide end-to-end security without relying on transport security.

**Pull request merged**